### ISSUE TITLE
clojure.spec Errors

### ISSUE THREAD

**Issue by nasser** [arcadia-unity/Arcadia#212]
Attempting to AOT compile Arcadia (as in #200) throws the following:

```
user=> clojure.lang.Compiler+CompilerException: System.Exception: :clojure.spec/unknown is not a fn, expected predicate fn
  at clojure/spec$dt__53213.invokeStatic (System.Object , System.Object , System.Object , System.Object ) [0x00000] in <filename unknown>:0 
  at clojure/spec$dt__53213.invoke (System.Object , System.Object , System.Object , System.Object ) [0x00000] in <filename unknown>:0 
  at clojure.spec$spec_impl$reify__53318__53322.conform_STAR_ (System.Object ) [0x00000] in <filename unknown>:0 
  at clojure/spec$conform__52838.invokeStatic (System.Object , System.Object ) [0x00000] in <filename unknown>:0 
  at clojure/spec$conform__52838.invoke (System.Object , System.Object ) [0x00000] in <filename unknown>:0 
  at clojure/spec$dt__53213.invokeStatic (System.Object , System.Object , System.Object , System.Object ) [0x00000] in <filename unknown>:0 
  at clojure/spec$dt__53213.invoke (System.Object , System.Object , System.Object , System.Object ) [0x00000] in <filename unknown>:0 
  at clojure/spec$dt__53213.invokeStatic (System.Object , System.Object , System.Object ) [0x00000] in <filename unknown>:0 
  at clojure/spec$dt__53213.invoke (System.Object , System.Object , System.Object ) [0x00000] in <filename unknown>:0 
  at clojure/spec$valid_QMARK___53220.invokeStatic (System.Object , System.Object ) [0x00000] in <filename unknown>:0 
  at clojure/spec$valid_QMARK___53220.invoke (System.Object , System.Object ) [0x00000] in <filename unknown>:0 
  at arcadia/internal/spec$loud_valid_QMARK___55819.invokeStatic (System.Object , System.Object ) [0x00000] in <filename unknown>:0 
  at arcadia/internal/spec$loud_valid_QMARK___55819.invoke (System.Object , System.Object ) [0x00000] in <filename unknown>:0 
  at arcadia/compiler$add_loadpath_extension_fnfn__57691__57703.invoke (System.Object ) [0x00000] in <filename unknown>:0 
  at clojure/core$update__37693.invokeStatic (System.Object , System.Object , System.Object ) [0x00000] in <filename unknown>:0 
  at clojure/core$update__37693.invoke (System.Object , System.Object , System.Object ) [0x00000] in <filename unknown>:0 
  at clojure.lang.Atom.swap (IFn f, System.Object arg1, System.Object arg2) [0x00000] in <filename unknown>:0 
  at clojure/core$swap_BANG___34369.invokeStatic (System.Object , System.Object , System.Object , System.Object ) [0x00000] in <filename unknown>:0 
  at clojure/core$swap_BANG___34369.invoke (System.Object , System.Object , System.Object , System.Object ) [0x00000] in <filename unknown>:0 
  at arcadia/compiler$add_loadpath_extension_fn__57706.invokeStatic (System.Object , System.Object ) [0x00000] in <filename unknown>:0 
  at arcadia/compiler$add_loadpath_extension_fn__57706.invoke (System.Object , System.Object ) [0x00000] in <filename unknown>:0 
  at clojure.lang.AFn.ApplyToHelper (IFn fn, ISeq argList) [0x00000] in <filename unknown>:0 
  at clojure.lang.AFn.applyTo (ISeq arglist) [0x00000] in <filename unknown>:0 
  at clojure.lang.CljCompiler.Ast.InvokeExpr.Eval () [0x00000] in <filename unknown>:0 , compiling: (leiningen.clj:153:2)
  at clojure.lang.CljCompiler.Ast.InvokeExpr.Eval () [0x00000] in <filename unknown>:0 
  at clojure.lang.RT.Compile (System.String dirName, System.String name, System.IO.TextReader rdr, System.String relativePath) [0x00000] in <filename unknown>:0 
  at clojure.lang.RT.Compile (System.IO.FileInfo cljInfo, System.String relativePath) [0x00000] in <filename unknown>:0 
  at clojure.lang.RT.load (System.String relativePath, Boolean failIfNotFound) [0x00000] in <filename unknown>:0 
  at clojure.lang.RT.load (System.String relativePath) [0x00000] in <filename unknown>:0 
  at clojure/core$loadfn__37639__37643.invoke () [0x00000] in <filename unknown>:0 
  at clojure/core$load__37646.invokeStatic (ISeq ) [0x00000] in <filename unknown>:0 
  at clojure/core$load__37646.doInvoke (System.Object ) [0x00000] in <filename unknown>:0 
  at clojure.lang.RestFn.invoke (System.Object arg1) [0x00000] in <filename unknown>:0 
  at clojure/core$load_one__37519.invokeStatic (System.Object , System.Object , System.Object ) [0x00000] in <filename unknown>:0 
  at clojure/core$load_one__37519.invoke (System.Object , System.Object , System.Object ) [0x00000] in <filename unknown>:0 
  at clojure/core$load_libfn__37557__37561.invoke () [0x00000] in <filename unknown>:0 
(2 stack frames omitted)
```

**Comment by timsgardner** [comment]
need the exact AOT invocation here plz

**Comment by timsgardner** [comment]
with this
```clojure
(arcadia.compiler/aot-namespaces
  "Assets/Arcadia/Compiled"
  '[clojure.core
    arcadia.core
    arcadia.repl
    arcadia.config
    arcadia.introspection
    arcadia.literals
    arcadia.packages
    arcadia.compiler])
```
it does throw that error but it also AOT's and everyone goes home. What's a couple errors here and there, between friends?

**Comment by nasser** [comment]
I was trying to AOT the whole shebang

```clojure
(compiler/aot-namespaces
  "Assets/Arcadia/Infrastructure"
  '[arcadia.core arcadia.repl arcadia.packages arcadia.linear
    arcadia.literals arcadia.config arcadia.compiler
    arcadia.internal.tracker arcadia.internal.thread arcadia.internal.test
    arcadia.internal.state arcadia.internal.spec arcadia.internal.seq-utils
    arcadia.internal.name-utils arcadia.internal.messages arcadia.internal.map-utils
    arcadia.internal.macro arcadia.internal.leiningen arcadia.internal.functions
    arcadia.internal.filewatcher-dummy arcadia.internal.file-system arcadia.internal.editor-interop
    arcadia.internal.components arcadia.internal.benchmarking arcadia.internal.asset-watcher
    arcadia.internal.array-utils arcadia.packages.data])
```

I do not believe it succeeded, but I would have to try again.

We need to look inside and understand.

![image](https://cloud.githubusercontent.com/assets/412966/23910611/b2161d82-08b0-11e7-85d2-0c14f0a84789.png)

**Comment by timsgardner** [comment]
closed as of a6b8fa7c2a23a731d2d2b01a66b1276ccdc984bc

**Comment by timsgardner** [comment]
(don't try to AOT arcadia.internal.seq_utils, it doesn't exist)
